packages/core/saved-objects/core-saved-objects-migration-server-internal/src/document_migrator/document_migrator.ts

```diff
@@ -180,10 +183,11 @@ export class DocumentMigrator implements VersionedTransformer {
     if (downgradeRequired(doc, typeMigrations?.latestVersion ?? {})) {
       const currentVersion = doc.typeMigrationVersion ?? doc.migrationVersion?.[doc.type];
       const latestVersion = this.migrations[doc.type].latestVersion[TransformType.Migrate];
+      const newerVersionModelError = createNewerModelVersionError(
```

Comment:
NIT: error creation should be within the `if` block, otherwise it will be created even when not used.